### ISSUE TITLE
1.3.1 and Lists and the notorious "or are available in text"

### ISSUE THREAD

**Issue by goodwitch** [w3c/wcag#1974]
The "or are available in text" clause in 1.3.1 is a giant loop hole.  For accuracy and consistency in interpretation of 1.3.1, it is important to clarify if "or are available in text" means that both of the following examples would technically meet (in the lowest of low ways) 1.3.1:

Example A:
`<p>I bought 3 things today: 1. bread, 2. butter, 3. eggs.</p>`

Example B:
`<p>I bought 3 things today: <br>1. bread <br>2. butter <br>3. eggs </p>`

One could easily argue that both example A and B meet 1.3.1 because the numbers (which are text) provide an indication of the list. 

Last, but not least, this paragraph in the Understanding further supports that both of these examples can be considered passing 1.3.1 (in the saddest of ways):

> There may also be cases where it may be a judgment call as to whether the relationships should be programmatically determined or be presented in text. However, when technologies support programmatic relationships, it is strongly encouraged that information and relationships be programmatically determined rather than described in text.

P.S.  Really, really not bringing this up to be annoying.  I care about this because accuracy and consistency across the world is important...and clarity in the SC language is key.   It would be awesome if that "or" clause could be put in parenthesis with a modifier that clear said, 

"1.3.1 Info and Relationships: Information, structure, and relationships conveyed through presentation can be programmatically determined **_or [are available in text_ IF (The technology in use does NOT provide the semantic structure to make the information and relationships conveyed through presentation programmatically determinable)].**

**Comment by mraccess77** [comment]
The criteria was created at a time when roles like headings were not available in Flash - or even in native Android apps.  Before ARIA expanded collapsed state was not available.  Today, most situations should and can be addressed programmatically - but not always. There are situations however, today such as with WebGL where some creative solutions exist but may not be able to use semantic HTML but roles and states can be communicated in ways that work albeit clunkily for many assistive technology.

**Comment by bruce-usab** [comment]
I agree with the sentiment that <q>or are available in text</q> is (nowadays) too much of a loophole.

**Comment by scottaohara** [comment]
Also tend to agree there is a loophole here, however I do hope that any update would not produce unnecessary failures.  

For instance, failing example A for not using a list would be incredibly heavy handed, as all users would be able to perceive the "list" in the same way - as a paragraph.  

Failing example B would make more sense (though opinions may vary on how necessary list semantics are in all cases) as the markup would create a visual representation of a semantic list – but without any of the exposed semantics of one.

**Comment by patrickhlauke** [comment]
agree with @scottaohara here. example A would be a super pedantic thing to fail. particularly since it sounds like there's no specific visual presentation here ... it's an inline, run-on paragraph of text.

**Comment by goodwitch** [comment]
I so agree with you @scottaohara and @patrickhlauke.  And every bit of me wants to fail Example B...but that dang "or" clause makes it hard to defend.  Sure...I know the intent of the SC...but that normative language is so loose.  (And I really do respect how fiendishly difficult it is to write SC that are objective, testable, and consistently measurable.)

**Comment by patrickhlauke** [comment]
example B, for me, is a very mild failure at best though. there's bigger fish to fry when it comes to sins commited by authors on their content. nobody is prevented from understanding the content, as outlined in the example, even as it is now...

**Comment by alastc** [comment]
If example B used something like:
`<p>I bought 3 things today: <span class="list-item">1. bread</span> <span class="list-item">2. butter</span> <span class="list-item">3. eggs</span> </p>`

With CSS that separated the items with a graphical indicator (e.g. bullets), then I'd fail that on 1.3.1. Otherwise nothing is being conveyed through presentation.

**Comment by patrickhlauke** [comment]
if they're styled/presented as a list, then i'd say it's a very mild actual fail of 1.3.1 for not using `<ol>` (as yes, "in text" is tells you a number for each item, but doesn't convey the whole "this entire thing is a list, and there are X total items, etc" that a proper `<ol>` conveys. and it gets particularly difficult if the items are quite wordy, as users then have to keep in mind they're in a list to begin with, etc)

**Comment by bruce-usab** [comment]
Just one person's anecdote, but I have given up on asserting that inserting the numbers manually (instead of semantically) fails 1.3.1.  In addition to what @patrickhlauke notes immediately above, the screen reading software provides a function to jump from list to list.  But I have yet to hear from any screen reader user that manual numbering gives them agita like it does me!

That said, I would fail @alastc example because there you have a developer going out of their way to code incorrectly.
